test-app/presenter/src/commonMain/kotlin/com/example/redwood/testing/presenter/TestApp.kt

```diff
@@ -30,62 +31,60 @@ import app.cash.redwood.ui.dp
 import com.example.redwood.testing.compose.Button
 import com.example.redwood.testing.compose.Text
 
+private val screens = buildMap<String, @Composable TestContext.() -> Unit> {
+  put("Repo Search") { RepoSearch(httpClient) }
+  put("UI Configuration") { UiConfigurationValues() }
+}
+
+@Stable
+class TestContext(
+  val httpClient: HttpClient,
+)
+
 @Composable
-fun TestApp(httpClient: HttpClient) {
-  val screen = remember { mutableStateOf<Screen?>(null) }
-  val activeScreen = screen.value
-  if (activeScreen == null) {
-    HomeScreen(screen)
+fun TestApp(context: TestContext) {
+  val screenKeyState = remember { mutableStateOf<String?>(null) }
+  val screenKey = screenKeyState.value
+  if (screenKey == null) {
+    ScreenList(screenKeyState)
   } else {
-    val onBack = { screen.value = null }
+    val onBack = { screenKeyState.value = null }
     BackHandler(onBack = onBack)
+
     Column(width = Fill, height = Fill) {
       Button("Back", onClick = onBack)
 
-      // TODO This should be a Box.
-      Column(
-        width = Fill,
-        horizontalAlignment = Stretch,
-        modifier = Modifier.grow(1.0).horizontalAlignment(Stretch),
-      ) {
-        activeScreen.Show(httpClient)
+      val content = screens[screenKey]
+      if (content == null) {
+        Text("No screen found with key '$screenKey'!")
```

Comment:
I thought about automatically falling back to home, but if we get `rememberSaveable`, you rename a key, and the app reloads, I want what happened to be slightly more clear.